app/controllers/candidate_interface/personal_details/immigration_entry_date_controller.rb

```diff
@@ -0,0 +1,47 @@
+module CandidateInterface
+  module PersonalDetails
+    class ImmigrationEntryDateController < CandidateInterfaceController
+      before_action :redirect_to_dashboard_if_submitted
+
+      def new
+        @form = ImmigrationEntryDateForm.build_from_application(current_application)
+      end
+
+      def create
+        @form = ImmigrationEntryDateForm.new(create_params)
+
+        if @form.save(current_application)
+          if LanguagesSectionPolicy.hide?(current_application)
+            redirect_to candidate_interface_personal_details_show_path
+          else
+            redirect_to candidate_interface_languages_path
+          end
+        else
+          track_validation_error(@form)
+          render :new
+        end
+      end
```

Comment:
So in these controllers there's no edit/update actions. Doesn't this mean that if they visit from the review/review application form pages they'll always be redirected back through the flow?

Reply:
that's true. I think if you change the answer to the _Do you have the right to work or study in the UK...?_ question you are going to have to go through the flow again. The only one that might be edited standalone that isn't already a last page in the flow is _What is your immigration status?_ I think. I will look at changing that one at least. Does that make sense?

Reply:
Yea that makes sense. I think we're struggling with these same tension on  the new GCSE flow. Maybe we need to co-opt this style for that flow too.